src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/AllowedCharsBitmap.cs

```diff
@@ -21,9 +21,9 @@ private AllowedCharsBitmap(uint[] allowedCharsBitmap)
         // Marks a character as allowed (can be returned unencoded)
         public void AllowCharacter(char c)
         {
-            uint codePoint = (uint)c;
-            int index = (int)(codePoint >> 5);
-            int offset = (int)(codePoint & 0x1FU);
+            int codePoint = c;
+            int index = codePoint >> 5;
```

Comment:
The argument starts as a char (16 bit unsigned). It's Max value will be 65K. When it's cast to an int the value is never negative (and fits in 32 bits just fine), so right shifts work as expected.